pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -558,9 +561,17 @@ func fitsResourceQuota(fName kueue.ResourceFlavorReference, rName corev1.Resourc
 		// ClusterQueue are preempted.
 		mode = Preempt
 	}
+	if cq.Preemption.BorrowWithinCohort != nil && cq.Preemption.BorrowWithinCohort.Policy != kueue.BorrowWithinCohortPolicyNever {
+		// when preemption with borrowing is enabled, we can succeeded admitting the
+		// workload if preemption is used.
+		if rQuota.BorrowingLimit != nil && val <= rQuota.Nominal+*rQuota.BorrowingLimit {
+			borrow = max(val-rQuota.Nominal, 0)
+			mode = Preempt
+		}
+	}
 	if rQuota.BorrowingLimit != nil && used+val > rQuota.Nominal+*rQuota.BorrowingLimit {
 		status.append(fmt.Sprintf("borrowing limit for %s in flavor %s exceeded", rName, fName))
-		return mode, 0, &status
+		return mode, borrow, &status
```

Comment:
if the val<rQuota.Nominal we will have 0 here, which means "no borrowing", is it what we want to return? if so, why?

Reply:
Yes, we want `false` (0) in that case, because this parameter means "borrowing required for the corresponding mode". I have also adjusted the comment.

So, if we return `Preempt` and `val<rQuota.Nominal` for a workload then it is possible to admit the workload without borrowing (just by preemption). 

Still, the workload might be later admitted using a combination of preemption and borrowing.

Reply:
ack